### DISCUSSION TITLE
How to run a ECS task once when cdk deploy

### DISCUSSION THREAD

**Discussion by Fitmavincent** [aws-samples/aws-cdk-examples#734] [Q&A]
I've been looking around a solution to run a ECS task once during ```cdk deploy```. I've tried ```EcsRunTask``` or using the state machine from Step functions. None of that triggers an run / execution of the task. Can anyone point me a direction on this?

``` typescript
// Run task
const runOnceTask = new EcsRunTask(this, "RunOnceTask", {
  integrationPattern: IntegrationPattern.RUN_JOB,
  cluster: cluster,
  taskDefinition: runOnceTaskDefinition,
  securityGroups: [roSg],
  assignPublicIp: false,
  subnets: vpc.selectSubnets({ subnetType: SubnetType.PRIVATE_WITH_NAT }),
  launchTarget: new EcsEc2LaunchTarget(),
    });
```

<img width="1466" alt="image" src="https://user-images.githubusercontent.com/7020812/190060612-5124012a-f427-4979-857c-898650b7b8aa.png">

```typescript
  // Create state machine
  const roStateMachine = new StateMachine(this, "RunOnceStateMachine", { definition: runOnceTask });
  // Start execution
  const execution = new StepFunctionsStartExecution(this, "RunOnceStartExecution", {
    stateMachine: roStateMachine,
    integrationPattern: IntegrationPattern.RUN_JOB,
    name: 'RuuOnceExecution'
  });
```
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/7020812/190060644-fdbf58e6-0255-43d7-9acd-90466003719d.png">

None of the above works for an execution

**Comment by geoynomous** [comment]
There is this https://github.com/pahud/cdk-fargate-run-task - but also that does not offer the required functionality ... I had the same problem - and because it's not solved, we've stopped CDK
